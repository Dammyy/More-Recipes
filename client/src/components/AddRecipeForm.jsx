import React, { PureComponent } from 'react';
import { Link } from 'react-router';
import { Field, reduxForm } from 'redux-form/immutable';
import PropTypes from 'prop-types';
/**
 *
 *
 * @export
 * @class AddRecipeForm
 * @extends {PureComponent}
 */
class AddRecipeForm extends PureComponent {
  /**
   *
   *
   *  @param {any} event
   * @returns {string} redirect
   * @memberOf AddRecipe
   */
  /** html component to render
   *
   *
   * @returns {void}
   *
   * @memberOf AddRecipeForm
   */
  render() {
    const { image, uploadImage } = this.props;
    return (
      <div className="container add-recipe-page">
        <div className="col-md-12 latest-recipes">
          <div className="text-left btn-buttons">
            <Link
              to="/catalog/manage"
              className="btn btn-info"
            >
              Back to My Recipes
            </Link>
          </div>
          <div className="panel panel-default">
            <div className="panel-heading">
              <h2 className="panel-title text-center">
                Add a Recipe!
              </h2>
            </div>
            <div className="panel-body">
              <form
                name="form"
                action=""
                onSubmit={this.props.handleSubmit}
                noValidate
              >
                <div className="form-group">
                  <label htmlFor="Title">Title</label>
                  <Field
                    name="title"
                    type="text"
                    id="Title"
                    component="input"
                    className="form-control"
                    placeholder="Title"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="ingredients">Ingredients</label>
                  <Field
                    name="ingredients"
                    component="input"
                    className="form-control"
                    placeholder="Ingredients e.g water, potato, rice"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="price">Details</label>
                  <Field
                    name="details"
                    component="textarea"
                    className="form-control"
                    rows="5"
                    placeholder="Recipe details"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="image">Image</label>
                  <div className="text-center dropup">
                    <button
                      id="button-upload"
                      type="button"
                      className="btn btn-danger"
                      onClick={() => uploadImage()}
                    >
                      Upload <span className="caret" />
                    </button>
                  </div>
                </div>
                <div className="form-group text-center upload-image">
                  <img
                    id="image"
                    className="img-responsive img-upload"
                    alt=""
                    src={image}
                  />
                </div>
                <button
                  type="submit"
                  className="btn btn-primary btn-block"
                >
                  Publish
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

AddRecipeForm.propTypes = {
  uploadImage: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  image: PropTypes.string.isRequired
};

export default reduxForm({ form: 'recipe' })(AddRecipeForm);
