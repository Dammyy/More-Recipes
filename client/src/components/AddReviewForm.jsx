import React, { PureComponent } from 'react';
import { Field, reset, reduxForm } from 'redux-form/immutable';
import PropTypes from 'prop-types';
/**
 *
 *
 * @export
 * @class AddRecipeForm
 * @extends {PureComponent}
 */
class AddReviewForm extends PureComponent {
  /** html component to render
   *
   *
   * @returns {void}
   *
   * @memberOf AddReviewForm
   */
  render() {
    return (

      <div id="recipe-display" className="col-md-12">
        <h2 className="panel-title text-center r-d-titles">
              Leave a review
        </h2>
        <div className="panel-body">
          <form
            name="form"
            action=""
            onSubmit={this.props.handleSubmit}
            noValidate
          >
            <div className="form-group">
              <Field
                name="review"
                type="text"
                id="review-input-field"
                component="textarea"
                className="form-control"
                placeholder="Leave a Review"
              />
            </div>
            <button
              type="submit"
              className="btn btn-primary btn-block"
            >
                Submit
            </button>
          </form>
        </div>
      </div>
    );
  }
}

AddReviewForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired
};

const afterSubmit = (result, dispatch) =>
  dispatch(reset('review'));

export default reduxForm({
  form: 'review',
  onSubmit: afterSubmit,
})(AddReviewForm);
