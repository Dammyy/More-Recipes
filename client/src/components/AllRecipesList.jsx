import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import RecipeHome from './RecipeHome';
import UserIsAuthenticated from '../utils/authWrapper';

const options = {
  authSelector: state => state.get('auth'),
  predicate: auth => auth.get('Authenticated'),
  wrapperDisplayName: 'authRecipeButtons',
  FailureComponent: null
};

const BtnAdd = UserIsAuthenticated(options)(() => (
  <div className="row text-left">
    <Link
      to="/catalog/add"
      className="btn btn-danger"
    >
  Add a new Recipe!
    </Link>
  </div>));
const BtnManageRecipes = UserIsAuthenticated(options)(() => (
  <div className="row text-left">
    <Link
      to="/catalog/manage"
      className="btn btn-publish"
    >
    Manage Recipes
    </Link>
  </div>));

const BtnFavorites = UserIsAuthenticated(options)(() => (
  <div className="row text-left">
    <Link
      to="/catalog/favorites"
      className="btn btn-publish"
    >
      My Favorites
    </Link>
  </div>));
/**
 *
 *
 * @class RecipeListHome
 * @extends {PureComponent}
 */
class AllRecipesList extends PureComponent {
  /**
   *
   *
   * @returns {void}
   *
   * @memberOf RecipeList
   */
  render() {
    const {
      recipes,
    } = this.props;
    return (
      <div>
        <BtnAdd />
        <BtnManageRecipes />
        <BtnFavorites />
        <div className="container scrollable">
          <div className="row">
            {
          recipes
            .map((recipe) => {
              return (
                <RecipeHome
                  {...recipe}
                  key={recipe.id}
                />
              );
            })
        }
          </div>
          <hr />
        </div>
      </div>
    );
  }
}
AllRecipesList.propTypes = {
  recipes: PropTypes.arrayOf(PropTypes.any).isRequired,
};

export default AllRecipesList;
