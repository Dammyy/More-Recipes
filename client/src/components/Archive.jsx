import React, { PureComponent } from 'react';
import { Link } from 'react-router';

export default class Home extends PureComponent {
  active (path) {
    if (this.props.location.pathname === path) {
      return 'active';
    }
  }
  render () {
    return (
<div>
        <nav className="navbar navbar-toggleable-md navbar-expand-md navbar-light bg-faded">
          <div className="header-logo">
            <a className="navbar-brand" href="index.html">
              <img src="img/logo.jpg" width={50} height={50} className="d-inline-block align-top" alt />
              <b>More Recipes<b>
                </b></b></a><b><b>
              </b></b></div><b><b>
              <button className="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
                <span className="navbar-toggler-icon" />
              </button>
              <div className="collapse navbar-collapse" id="navbarNav">
                <ul className="navbar-nav">
                  <li className="nav-item">
                    <a className="nav-link" href="catalog.html">Catalog</a>
                  </li>
                  <li className="nav-item">
                    <a className="nav-link" href="fav.html">Favorites</a>
                  </li>
                  <li className="nav-item" className={this.active('/manage')}>
                    <Link to="/manage" className="nav-link">Manage recipes</Link>
                  </li>
                  <li className="nav-item" className={this.active('/catalog')}>
                    <Link to="/catalog" className="nav-link">Catalog</Link>
                  </li>
                  <li className="nav-item">
                    <a className="nav-link" href="profile.html">Profile</a>
                  </li>
                  <li className="nav-item">
                    <a className="nav-link" href="signup.html">Sign up</a>
                  </li>
                  <li className="nav-item">
                    <a className="nav-link" href="signin.html">Sign in</a>
                  </li>
                </ul>
				
              </div>
            </b></b></nav><b><b>
            <div className="container-fluid">
              <div className="row">
                <div className="col-md-9 catalog-left">
                  <div id="catalog-search-form">
                    <form className="form-inline">
                      <input className="form-control mr-sm-2" type="text" placeholder="Search" />
                      <button className="btn btn-outline-success my-2 my-sm-0" type="submit">Search</button>
                    </form> 
                  </div>
                  <div className="col-md-12 latest-recipes">
                  <div className="row">
                    {this.props.children}
                </div>
                </div>
                </div>
                <div className="col-md-3 catalog-right">
                  <h3 id="latest-h3">Popular Recipes</h3>
                  <div id="popular">
                    <div id="popular-title"><a href="#"> Awesome chicken </a></div>
                    <div id="popular-votes"><li><i className="fa fa-thumbs-o-up" aria-hidden="true"> 200</i> </li><li><i className="fa fa-thumbs-o-down" aria-hidden="true"> 50</i></li> </div>
                  </div>
                  <div id="popular">
                    <div id="popular-title"><a href="#"> Grilled turkey </a></div>
                    <div id="popular-votes"><li><i className="fa fa-thumbs-o-up" aria-hidden="true"> 200</i> </li><li><i className="fa fa-thumbs-o-down" aria-hidden="true"> 50</i></li> </div>
                  </div>
                  <div id="popular">
                    <div id="popular-title"><a href="#"> Awesome chicken </a></div>
                    <div id="popular-votes"><li><i className="fa fa-thumbs-o-up" aria-hidden="true"> 200</i> </li><li><i className="fa fa-thumbs-o-down" aria-hidden="true"> 50</i></li> </div>
                  </div>
                </div>
              </div>
            </div>


            <footer className="footer">
              <div className="container">
                <div className="row">
                  <div className="col-md-6">
                    <a href="https://github.com/Dammyy"> <img src="img/github.png" /></a>
                  </div>
                  <div className="col-md-6">
                    <span className="text-muted">Copyright Â© More Recipes!</span>
                  </div>
                </div>
              </div>
            </footer>
            
          </b></b></div>
          
    );
  }
}
