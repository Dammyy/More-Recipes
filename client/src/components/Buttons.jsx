import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import UserIsAuthenticated from '../utils/authWrapper';

const options = {
  authSelector: state => state.get('auth'),
  predicate: auth => auth.get('Authenticated'),
  wrapperDisplayName: 'authRecipeButtons',
  FailureComponent: null
};

export const BtnAdd = (() => (
  <Link
    to="/catalog/add"
    className="btn btn-publish btn-manage"
  >
    <i className="fa fa-pencil-square-o" aria-hidden="true" /> Add Recipe
  </Link>
));
const AuthenticatedBtnAdd = UserIsAuthenticated(options)(BtnAdd);

export const BtnManageRecipes = (() => (
  <Link
    to="/catalog/manage"
    className="btn btn-publish btn-manage"
  >
    <i className="fa fa-list-alt" aria-hidden="true" /> Manage Recipes
  </Link>
));
const AuthenticatedBtnManageRecipes =
UserIsAuthenticated(options)(BtnManageRecipes);

export const BtnFavorites = (() => (
  <Link
    to="/catalog/favorites"
    className="btn btn-publish btn-manage"
  >
    <i className="fa fa-heart" /> My Favorites
  </Link>
));
const AuthenticatedBtnFavorites = UserIsAuthenticated(options)(BtnFavorites);

const BtnHome = (() => (
  <Link
    to="/"
    className="btn btn-info btn-manage"
  >
    <i className="fa fa-home" /> Home
  </Link>
));

const BtnCatalog = (() => (
  <Link
    to="/catalog"
    className="btn btn-publish btn-manage"
  >
    <i className="fa fa-list-alt" aria-hidden="true" /> Catalog
  </Link>
));

export const BtnProfile = (() => (
  <Link
    to="/profile"
    className="btn btn-publish btn-manage"
  >
    <i className="fa fa-user-o" aria-hidden="true" /> My Profile
  </Link>
));
const AuthenticatedBtnProfile = UserIsAuthenticated(options)(BtnProfile);

export const BtnEdit = ((props) => {
  return (
    <Link
      className="btn btn-publish btn-manage"
      to={`edit/${props.id}`}
      params={{ id: props.id }}
    >
      Edit
    </Link >);
});
const AuthenticatedBtnEdit = UserIsAuthenticated(options)(BtnEdit);

const BtnCancel = (() => (
  <Link
    to="/catalog/manage"
    className="btn btn-danger btn-manage"
  >
    <i className="fa fa-times" aria-hidden="true" /> Cancel
  </Link>
));

const BtnCurrent = (() => (
  'My Recipes'
));

BtnEdit.propTypes = {
  id: PropTypes.number.isRequired
};

export {
  AuthenticatedBtnAdd,
  AuthenticatedBtnManageRecipes,
  BtnHome,
  BtnCatalog,
  AuthenticatedBtnFavorites,
  AuthenticatedBtnProfile,
  AuthenticatedBtnEdit,
  BtnCancel,
  BtnCurrent
};
