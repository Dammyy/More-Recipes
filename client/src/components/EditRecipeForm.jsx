import React, { PureComponent } from 'react';
import { Link } from 'react-router';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form/immutable';
import PropTypes from 'prop-types';
/**
 *
 *
 * @export
 * @class EditRecipeForm
 * @extends {PureComponent}
 */
class EditRecipeForm extends PureComponent {
  /**
   *
   * @returns {void}
   *
   * @memberOf EditRecipeForm
   */
  componentDidMount() {
    if (this.props.recipes.length < 1) {
      this.props.redirectUser('/catalog');
    }
  }
  /**
   *
   *
   *  @param {any} event
   * @returns {string} redirect
   * @memberOf AddRecipe
   */
  /** html component to render
   *
   *
   * @returns {void}
   *
   * @memberOf AddRecipeForm
   */
  render() {
    const { image, uploadImage } = this.props;
    return (
      <div className="col-md-9 catalog-left">
        <div className="text-left">
          <Link to="/catalog" className="btn btn-info">Back to Catalog</Link>
        </div>
        <div className="panel panel-default">
          <div className="panel-heading">
            <h2 className="panel-title text-center">
              Edit Recipe
            </h2>
          </div>
          <div className="panel-body">
            <form
              name="form"
              action=""
              onSubmit={this.props.handleSubmit}
              noValidate
            >
              <div className="form-group">
                <label htmlFor="Title">Title</label>
                <Field
                  name="title"
                  type="text"
                  id="Title"
                  component="input"
                  className="form-control"
                  placeholder="Title"
                />
              </div>
              <div className="form-group">
                <label htmlFor="ingredients">Ingredients</label>
                <Field
                  name="ingredients"
                  component="input"
                  className="form-control"
                  placeholder="Ingredients e.g water, potato, rice"
                />
              </div>
              <div className="form-group">
                <label htmlFor="price">Details</label>
                <Field
                  name="details"
                  component="textarea"
                  className="form-control"
                  rows="5"
                  placeholder="Recipe details"
                />
              </div>
              <div className="form-group">
                <label htmlFor="image">Image</label>
                <div className="text-center dropup">
                  <button
                    id="button-upload"
                    type="button"
                    className="btn btn-danger"
                    onClick={() => uploadImage()}
                  >
                    Upload <span className="caret" />
                  </button>
                </div>
              </div>
              <div className="form-group text-center">
                <img
                  id="image"
                  className="img-responsive img-upload"
                  alt=""
                  src={image}
                />
              </div>
              <button
                type="submit"
                className="btn btn-primary btn-block"
              >
                Publish
              </button>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

EditRecipeForm.propTypes = {
  uploadImage: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  image: PropTypes.string.isRequired,
  recipes: PropTypes.arrayOf(PropTypes.any).isRequired,
  redirectUser: PropTypes.func.isRequired
};

const mapStateToProps = (state, ownProps) => {
  const { recipes, id } = ownProps;
  const recipe = recipes.filter(recp => recp.id === parseInt(id, 10))[0];
  return {
    initialValues: recipe,
  };
};

const connectReduxForm = reduxForm({
  form: 'updateRecipe',
  enableReinitialize: true,
})(EditRecipeForm);

export default connect(mapStateToProps)(connectReduxForm);
