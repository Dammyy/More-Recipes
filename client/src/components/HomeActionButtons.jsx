import React, { PureComponent } from 'react';
import { Link } from 'react-router';
import PropTypes from 'prop-types';
import AuthCheck from '../utils/authWrapper';

/**
 *
 *
 * @class HomeActionButtons
 * @extends {PureComponent}
 */
class HomeActionButtons extends PureComponent {
  /**
   *
   * @memberOf HomeActionButtons
   * @returns {void}
   *
   */
  render() {
    const { firstName, logout } = this.props;
    return (
      <div className="publish-recipe">
        <h2>You are logged in as {firstName}</h2>
        <p>
          <Link
            to="/catalog"
            className="btn btn-post-recipe"
          >
            <i
              className="fa fa-eye"
              aria-hidden="true"
            /> Browse
          </Link>
          <Link
            to="/catalog/add"
            className="btn btn-post-recipe"
          >
            <i
              className="fa fa-pencil-square-o"
              aria-hidden="true"
            /> New Recipe
          </Link>
          <button
            className="btn btn-logout"
            onClick={logout}
          >
            <i className="fa fa-power-off" aria-hidden="true" /> Logout
          </button>
        </p>
      </div>
    );
  }
}

const options = {
  authSelector: state => state.get('auth'),
  predicate: auth => auth.get('Authenticated'),
  wrapperDisplayName: 'authAddRecipe',

  FailureComponent: () => {
    return (
      <div className="publish-recipe" role="group" aria-label="...">
        <Link
          to="/catalog"
          className="btn btn-signup"
        >
          <i
            className="fa fa-eye"
            aria-hidden="true"
          /> Browse
        </Link>
        <Link to="/signup" className="btn btn-signup">Sign Up</Link>
        <Link to="/login" className="btn btn-signin">Login</Link>
      </div>
    );
  }
};

HomeActionButtons.propTypes = {
  firstName: PropTypes.string.isRequired,
  logout: PropTypes.func.isRequired,
};
export default AuthCheck(options)(HomeActionButtons);
