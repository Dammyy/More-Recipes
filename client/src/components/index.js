import Home from './Home';
import Welcome from './Welcome';
import Archive from './Archive';
import AddRecipeForm from './AddRecipeForm';
import EditRecipeForm from './EditRecipeForm';
import Recipe from './Recipe';
import RecipesList from './RecipesList';
import Login from './Login';
import Signup from './Signup';
import ViewRecipe from './ViewRecipe';
import NotFoundPage from './NotFoundPage';
import AllRecipesList from './AllRecipesList';
import MyFavoriteRecipes from './MyFavoriteRecipes';
import AddReviewForm from './AddReviewForm';
import Reviews from './Reviews';

export { Home, Welcome, AddRecipeForm, Recipe,
  RecipesList, Archive, Login, Signup, EditRecipeForm,
  ViewRecipe, NotFoundPage, AllRecipesList, MyFavoriteRecipes,
  AddReviewForm, Reviews };
