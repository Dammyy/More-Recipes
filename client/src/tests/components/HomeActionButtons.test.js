import React from 'react';
import { shallow } from 'enzyme';
import { HomeActionButtons } from '../../components/HomeActionButtons';

describe('Navbar component', () => {
  describe('Snapshot', () => {
    const props = {
      firstName: '',
      logout: jest.fn()
    };

    test('component should fully render', () => {
      const component = shallow(<HomeActionButtons {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
});
