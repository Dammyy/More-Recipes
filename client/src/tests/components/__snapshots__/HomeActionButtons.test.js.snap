// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Snapshot component should fully render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeActionButtons
    firstName=""
    logout={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          You are logged in as 
          
        </h2>,
        <p>
          <Link
            className="btn btn-post-recipe"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/catalog"
          >
            <i
              aria-hidden="true"
              className="fa fa-eye"
            />
             Browse
          </Link>
          <Link
            className="btn btn-post-recipe"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/catalog/add"
          >
            <i
              aria-hidden="true"
              className="fa fa-pencil-square-o"
            />
             New Recipe
          </Link>
          <button
            className="btn btn-logout"
            onClick={[MockFunction]}
          >
            <i
              aria-hidden="true"
              className="fa fa-power-off"
            />
             Logout
          </button>
        </p>,
      ],
      "className": "publish-recipe",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "You are logged in as ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          "You are logged in as ",
          "",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="btn btn-post-recipe"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/catalog"
            >
              <i
                aria-hidden="true"
                className="fa fa-eye"
              />
               Browse
            </Link>,
            <Link
              className="btn btn-post-recipe"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/catalog/add"
            >
              <i
                aria-hidden="true"
                className="fa fa-pencil-square-o"
              />
               New Recipe
            </Link>,
            <button
              className="btn btn-logout"
              onClick={[MockFunction]}
            >
              <i
                aria-hidden="true"
                className="fa fa-power-off"
              />
               Logout
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  aria-hidden="true"
                  className="fa fa-eye"
                />,
                " Browse",
              ],
              "className": "btn btn-post-recipe",
              "onlyActiveOnIndex": false,
              "style": Object {},
              "to": "/catalog",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "className": "fa fa-eye",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Browse",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  aria-hidden="true"
                  className="fa fa-pencil-square-o"
                />,
                " New Recipe",
              ],
              "className": "btn btn-post-recipe",
              "onlyActiveOnIndex": false,
              "style": Object {},
              "to": "/catalog/add",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "className": "fa fa-pencil-square-o",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " New Recipe",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  aria-hidden="true"
                  className="fa fa-power-off"
                />,
                " Logout",
              ],
              "className": "btn btn-logout",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "className": "fa fa-power-off",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Logout",
            ],
            "type": "button",
          },
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            You are logged in as 
            
          </h2>,
          <p>
            <Link
              className="btn btn-post-recipe"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/catalog"
            >
              <i
                aria-hidden="true"
                className="fa fa-eye"
              />
               Browse
            </Link>
            <Link
              className="btn btn-post-recipe"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/catalog/add"
            >
              <i
                aria-hidden="true"
                className="fa fa-pencil-square-o"
              />
               New Recipe
            </Link>
            <button
              className="btn btn-logout"
              onClick={[MockFunction]}
            >
              <i
                aria-hidden="true"
                className="fa fa-power-off"
              />
               Logout
            </button>
          </p>,
        ],
        "className": "publish-recipe",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "You are logged in as ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            "You are logged in as ",
            "",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="btn btn-post-recipe"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/catalog"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-eye"
                />
                 Browse
              </Link>,
              <Link
                className="btn btn-post-recipe"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/catalog/add"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-pencil-square-o"
                />
                 New Recipe
              </Link>,
              <button
                className="btn btn-logout"
                onClick={[MockFunction]}
              >
                <i
                  aria-hidden="true"
                  className="fa fa-power-off"
                />
                 Logout
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <i
                    aria-hidden="true"
                    className="fa fa-eye"
                  />,
                  " Browse",
                ],
                "className": "btn btn-post-recipe",
                "onlyActiveOnIndex": false,
                "style": Object {},
                "to": "/catalog",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-eye",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " Browse",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <i
                    aria-hidden="true"
                    className="fa fa-pencil-square-o"
                  />,
                  " New Recipe",
                ],
                "className": "btn btn-post-recipe",
                "onlyActiveOnIndex": false,
                "style": Object {},
                "to": "/catalog/add",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-pencil-square-o",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " New Recipe",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    aria-hidden="true"
                    className="fa fa-power-off"
                  />,
                  " Logout",
                ],
                "className": "btn btn-logout",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-power-off",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " Logout",
              ],
              "type": "button",
            },
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
